Framework/Kernel/test/LambdaValidatorTest.h

```diff
@@ -0,0 +1,70 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidKernel/IValidator.h"
+#include "MantidKernel/LambdaValidator.h"
+
+using Mantid::Kernel::IValidator_sptr;
+using Mantid::Kernel::LambdaValidator;
+
+class LambdaValidatorTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static LambdaValidatorTest *createSuite() { return new LambdaValidatorTest(); }
+  static void destroySuite(LambdaValidatorTest *suite) { delete suite; }
+
+  void testEmptyLambda() {
+    LambdaValidator<int> validator([](int x) {
+      // trivial if to remove unused warning
+      if (x == 0) {
```

Comment:
You should be able to use a warning suppressor, as done here:
https://github.com/mantidproject/mantid/pull/16854/files

Reply:
https://developer.arm.com/documentation/dui0491/c/Compiler-specific-Features/--attribute----unused---variable-attribute?lang=en#:~:text=for%20your%20feedback-,__attribute__((unused))%20variable%20attribute,if%20it%20is%20not%20used.

Reply:
It turned out that GNU_UNUSED_FUNCTION is unused. Also, this directive can be used in functions and variables so its name is misleading. Can I just rename it here or it should be in a separate PR?

Reply:
Agree with the misleading naming, I think rename it here with a decent commit message.

Reply:
Ok